### PR TITLE
Type evolved storage.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi Max!

TL;DR:

```
 942 TEST(TypeEvolutionTest, StorageInnerTypes) {
 943   // Test the plain `Name` type goes through.
 944   typename SchemaOriginalStorage::Name foo;
 945   foo.first = "Ayn";
 946   foo.last = "Rand";
 947   {
 948     typename SchemaOriginalStorage::Name bar;
 949     current::type_evolution::Evolve<SchemaOriginalStorage, typename SchemaOriginalStorage::Name>::template Go<
 950         SchemaOriginalStorage>(foo, bar);
 951     EXPECT_EQ("{\"first\":\"Ayn\",\"last\":\"Rand\"}", JSON(bar));
 952   }
 953   {
 954     typename SchemaOriginalStorage::Name baz;
 955     current::type_evolution::Evolve<SchemaOriginalStorage,
 956                                     typename SchemaOriginalStorage::Name,
 957                                     current::type_evolution::OriginalStorageToOriginalStorageEvolutor>::
 958         template Go<SchemaOriginalStorage>(foo, baz);
 959     EXPECT_EQ("{\"first\":\"AYN\",\"last\":\"RAND\"}", JSON(baz));
 960   }
 961   {
 962     typename SchemaModifiedStorage::Name meh;
 963     current::type_evolution::Evolve<SchemaOriginalStorage,
 964                                     typename SchemaOriginalStorage::Name,
 965                                     current::type_evolution::OriginalStorageToModifiedStorageEvolutor>::
 966         template Go<SchemaModifiedStorage>(foo, meh);
 967     EXPECT_EQ("{\"full\":\"Rand, A\"}", JSON(meh));
 968   }
 969 }
```

```
$ cat golden/storage_pre_evolution.txt 
{"index":0,"us":1}	{"meta":{"timestamp":0,"fields":{}},"mutations":[{"PersistedUserUpdated":{"data":{"first":"KARL","last":"MARX","key":"karl"}},"":"T9201121062548426382"}]}
{"index":1,"us":3}	{"meta":{"timestamp":2,"fields":{}},"mutations":[{"PersistedUserUpdated":{"data":{"first":"DIMA","last":"KOROLEV","key":"dima"}},"":"T9201121062548426382"},{"PersistedUserUpdated":{"data":{"first":"MAX","last":"ZHUROVICH","key":"max"}},"":"T9201121062548426382"}]}

$ cat golden/storage_post_evolution.txt 
{"index":0,"us":1}	{"meta":{"timestamp":0,"fields":{}},"mutations":[{"PersistedUserUpdated":{"data":{"full":"Marx, K","key":"karl"}},"":"T9205376179718671976"}]}
{"index":1,"us":3}	{"meta":{"timestamp":2,"fields":{}},"mutations":[{"PersistedUserUpdated":{"data":{"full":"Korolev, D","key":"dima"}},"":"T9205376179718671976"},{"PersistedUserUpdated":{"data":{"full":"Zhurovich, M","key":"max"}},"":"T9205376179718671976"}]}

Главное, ребята, сердцем не стареть.

### CODE REVIEW

**Comment by dkorolev** [comment]
@mzhurovich -- should be good to look through.

I'm beginning to like your suggetsions of removing `using T... = ...` from autogenerated schema. Would probably remove non-explicitly-called `Variant<>`-s too.

Thanks,
Dima

File: TypeSystem/Evolution/type_evolution.h [hunk 1]
```diff
@@ -61,6 +68,32 @@ struct Evolve;
 #include "../primitive_types.dsl.h"
 #undef CURRENT_DECLARE_PRIMITIVE_TYPE
 
+// Boilerplate default generic evolutor for `std::vector<T>`.
+template <typename FROM_NAMESPACE, typename EVOLUTOR, typename VECTOR_ELEMENT_TYPE>
+struct Evolve<FROM_NAMESPACE, std::vector<VECTOR_ELEMENT_TYPE>, EVOLUTOR> {
+  template <typename INTO, typename OUTPUT>
+  static void Go(const std::vector<VECTOR_ELEMENT_TYPE>& from, OUTPUT& into) {
+    into.resize(from.size());
```
Comment (mzhurovich) [inline TypeSystem/Evolution/type_evolution.h]: s/resize/reserve/?

Comment (dkorolev) [inline TypeSystem/Evolution/type_evolution.h]: Our `Evolve<...>::Go(from, into)` requires an existing placeholder, and `.resize()` allocates it and makes it accessible. The alternative would be to `return` the evolved objects and `emplace_back` them, but we decided `OUTPUT& output` as the 2nd parameter to `Evolve<...>::Go()` is a cleaner way

**Comment by dkorolev** [comment]
Го https://github.com/C5T/Current/pull/529, я создал.
